man/mustache.5.ron

```diff
@@ -286,6 +411,80 @@ Can be thought of as a single, expanded template:
     {{/names}}
 
 
+### Blocks
+
+A block begins with a dollar and ends with a slash. That is, `{{$title}}`
+begins a "title" block and `{{/title}}` ends it.
+
+Blocks mark parts of the template that may be overridden. This can be done
+with a block of the same name within a parent section in the calling
+template (see **Parents** below). If not overridden, the contents of a
+block render just as if the `{{$title}}` and `{{/title}}` tags weren't
+there.
+
+Blocks could be thought of as template parameters or as inline partials
+that may be passed to another template. They are part of the optional
+inheritance extension.
+
+Template `article.mustache`:
+
+    <h1>{{$title}}The News of Today{{/title}}</h1>
+    {{$body}}
+    <p>Nothing special happened.</p>
+    {{/body}}
+
+Output:
+
+    <h1>The News of Today</h1>
+    <p>Nothing special happened.</p>
+
+
+### Parents
+
+A parent begins with a less than sign and ends with a slash. That is,
+`{{<article}}` begins an "article" parent and `{{/article}}` ends it.
+
+Like an `{{>article}}` partial, a parent lets you expand another template
+inside the current one. Unlike a partial, a parent also lets you override
+blocks of the other template.
+
+Blocks within a parent can again be overridden by another including
+template. Other content within a parent is ignored, like comments.
+
+Template:
+
+    {{<article}}
+      Never shown
```

Comment:
IMHO, this could benefit from some clarification I wasn't getting why this is supposed to never show. After a lot of head-scratching, it seems like you "call" a parent, and then the content of the parent gets included in the template. You can override the blocks by re-defining them, but everything which is not a block redefinition gets ignored. That seems a bit confusing without clarification. I'm not even sure it works this way.

How about expanding the second paragraph as this:

> Like an `{{>article}}` partial, a parent lets you expand another template
> inside the current one. Unlike a partial, a parent also lets you override
> blocks of the other template. The parent can only override blocks, and
> anything else outside a block is discarded and not rendered.

Reply:
Good suggestion, thanks!

Everything between the `{{<parent}}{{/parent}}` tags is considered a comment except for the blocks.

Reply:
@suy At second thought, it seems that the next paragraph already explains this:

https://github.com/mustache/mustache/blob/d46a7c5411bcefe85e7205c3dce037cac151ef0c/man/mustache.5.ron#L451-L452

Doesn't that sufficiently address the confusion?

Reply:
Now that you point it at it, and with the knowledge of what it does, yes. I read it "other content is ignored like if it were a comment". Before I was reading it like "other content, like comments, is ignored". Feel free to ignore my remark. I'm just a person who is not used to use templates, so maybe my experience doesn't translate to others. And it seems I got the meaning right anyway, just without much confidence. Thanks!

Reply:
Welcome. Out of curiosity, how does a person who is not used to using templates end up reviewing a pull request for the manpage of a templating language? :-)

Reply:
A bit of a niche case... :) There is a [scripting language for a very old video game engine](https://www.pocketplane.net/tutorials/simscript.html) with pretty severe limitations. I did a first attempt at making a script for it with some minimal abstraction using the C/C++ preprocessor. But someone suggested using a template language, and here I am, shopping for one. I remember I've heard of mustache eons ago, and despite that the website and the Ruby implementation are not in a great shape, I've not liked the alternatives. Luckily I've found [a C++ one](https://github.com/jamboree/bustache) that seems it will be fine (supports the inheritance part that I was trying to understand).